test/integration/apiserver/apply/apply_test.go

```diff
@@ -244,6 +244,117 @@ func TestNoOpUpdateSameResourceVersion(t *testing.T) {
 	}
 }
 
+// TestNoOpApplyWithEmptyMap
+func TestNoOpApplyWithEmptyMap(t *testing.T) {
+	client, closeFn := setup(t)
+	defer closeFn()
+
+	deploymentName := "no-op"
+	deploymentsResource := "deployments"
+	deploymentBytes := []byte(`{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "` + deploymentName + `",
+			"labels": {
+				"app": "nginx"
+			}
+		},
+		"spec": {
+			"replicas": 1,
+			"selector": {
+				"matchLabels": {
+					"app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"annotations": {},
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name": "nginx",
+						"image": "nginx:1.14.2",
+						"ports": [{
+							"containerPort": 80
+						}]
+					}]
+				}
+			}
+		}
+	}`)
+
+	_, err := client.AppsV1().RESTClient().Patch(types.ApplyPatchType).
+		Namespace("default").
+		Param("fieldManager", "apply_test").
+		Resource(deploymentsResource).
+		Name(deploymentName).
+		Body(deploymentBytes).
+		Do(context.TODO()).
+		Get()
+	if err != nil {
+		t.Fatalf("Failed to create object: %v", err)
+	}
+
+	// Sleep for one second to make sure that the times of each update operation is different.
+	time.Sleep(1 * time.Second)
```

Comment:
not necessary here, is it?

Reply:
Unfortunately it is..  The timestamp is 1s granularity, and results in this test flaking if we don't to a sleep. The other tests for nops are also doing this wait..

Reply:
Ack. Maybe worth to mention: "This is necessary to trigger the bug."

Reply:
SG, I've expanded the comment to spell it out.  We could avoid this with a fake clock, but for now, this should should hold up.

Reply:
> // This sleep is necessary to trigger the bug consistently because the time field in managedFields has
> // 1 second granularity and if both apply requests happen during the same second, this test would flake.

ideally, this should reference a ticket or describe "the bug" in question.

not sure i understand the quantization of the managed field in question here, but is it sufficient to just delay the second call by 1 second so that they are not within one second? what if the CI setup is slow and it ends up being exactly 2 seconds. would that flake?

Reply:
Rephrasing this as: 

```
// This sleep is necessary to consistently produce different timestamps because the time field in managedFields has
// 1 second granularity and if both apply requests happen during the same second, this test would flake.
```

No, 2 seconds wouldn't flake. The only thing that needs to happen to prevent flakes is prevent the two apply requests from being stamped with the same exact timestamp.  Since the timestamps have 1s granularity, this should be sufficient (modulo any non-contentiousness in clocks).